src/parser.c

```diff
@@ -1879,6 +1885,18 @@ int eval_expression_imm(opcode_t op, int op1, int op2)
     case OP_rshift:
         res = op1 >> op2;
         break;
+    case OP_log_and:
+        res = op1 && op2;
+        break;
+    case OP_log_or:
+        res = op1 || op2;
+        break;
+    case OP_eq:
+        res = op1 == op2 ? 1 : 0;
```

Comment:
If possible, directly rewrite it as `res = op1 == op2` to avoid generating branch instructions.